### DISCUSSION TITLE
[Question][Hw4] Does capability update?

### DISCUSSION THREAD

**Discussion by ICubE01** [prosyslab-classroom/cs348-information-security#706] [Q&A]
Name: Jeho Yeon

What is the result of the scenario below?
It is modification of scenario2.c, remove() added.
```
const enum access_control_t access_control = Capability;

void init() {
  create_subject("attacker");
  create_subject("deputy");
  create_object("secret_file");
  create_object("deputy");
  enter(Read, "attacker", "secret_file");
  enter(Execute, "attacker", "deputy");
}

void attacker() {
  int fd = open("secret_file", Read);
  remove(Read, "attacker", "secret_file");
  execute_with_capability(fd, "deputy");
}

void deputy(int fd) { read_with_capability(fd, "secret_file"); }
```

After remove(), fd loses its capability or not?
Similarly, after confer(), could file descriptor gain a capability?

**Comment by ICubE01** [comment]
Additionally, is it right that execute_with_capability() doesn't use capability to execute a process?
Is capability only used in read_with_capability()?

**Comment by sujin0529** [comment]
Hi,

1. Same as Linux. If you think about the `file descriptor`, you'll find the answer quickly.
2. You are right in your understanding.

Thanks.
